### PR TITLE
Add search pipeline metrics to Nodes Stats API

### LINKED ISSUE #4511: [DOC] Search Pipeline Metrics
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [x] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 Relating to [Search pipeline metrics (at least on par w/ ingestion pipeline metrics).#6723](https://github.com/opensearch-project/OpenSearch/issues/6723), we want to add a page under Search Pipelines that show what metrics we are exposing. 

This affects 2.9 only as this feature is new to 2.9. 

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Fixes #4511 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kolchfa-aws** [comment]
@noCharger This PR is ready for tech review. Thanks!

**Review by noCharger — APPROVED** [review]
LGTM, thanks!

**Review by vagimeli — APPROVED** [review]
No changes. LGTM

File: _api-reference/nodes-apis/nodes-stats.md [hunk 1]
```diff
@@ -936,6 +1002,34 @@ pipelines._id_.time_in_millis | Integer | The total amount of time for preproces
...
+total_request.failed | Integer | The total number of failed search request processor executions.
+total_response | Object | Cumulative statistics related to all search response processors.
+total_response.count | Integer | The total number of search response processor executions.
+total_response.time_in_millis | Integer | The total amount of time for all search response processor executions, in milliseconds.
+total_response.current | Integer | The total number of search response processor executions currently in progress.
+total_response.failed | Integer | The total number of failed search response processor executions.
+pipelines | Object | Search pipeline statistics. Each pipeline is a nested object specified by its ID with the properties listed in the following rows. If a processor has a `tag`, statistics for the processor are provided in the object with the name `<processor_type>:<tag>` (for example, `filter_query:abc`). Statistics for all processors of the same type that do not have a `tag` are aggregated and provided in the object with the name `<processor-type>` (for example, `filter_query`).
```
Comment (natebower) [inline _api-reference/nodes-apis/nodes-stats.md:1021]:
```suggestion
pipelines | Object | Search pipeline statistics. Each pipeline is a nested object specified by its ID, with the properties listed in the following rows. If a processor has a `tag`, statistics for the processor are provided in the object with the name `<processor_type>:<tag>` (for example, `filter_query:abc`). Statistics for all processors of the same type that do not have a `tag` are aggregated and provided in the object with the name `<processor-type>` (for example, `filter_query`).
```

File: _api-reference/nodes-apis/nodes-stats.md [hunk 2]
```diff
@@ -936,6 +1002,34 @@ pipelines._id_.time_in_millis | Integer | The total amount of time for preproces
...
+pipelines._id_.request.count | Integer | The number of search request processor executions by the search pipeline.
```
Comment (natebower) [inline _api-reference/nodes-apis/nodes-stats.md:1022]:
```suggestion
pipelines._id_.request.count | Integer | The number of search request processor executions performed by the search pipeline.
```

File: _api-reference/nodes-apis/nodes-stats.md [hunk 3]
```diff
@@ -936,6 +1002,34 @@ pipelines._id_.time_in_millis | Integer | The total amount of time for preproces
...
+pipelines._id_.request.time_in_millis | Integer | The total amount of time for search request processor executions in the search pipeline, in milliseconds.
+pipelines._id_.request.current | Integer | The number of search request processor executions currently in progress for the search pipeline.
+pipelines._id_.request.failed | Integer | The number of failed search request processor executions for the search pipeline.
+pipelines._id_.request_processors | Array of objects | Statistics for the search request processors. Includes the total number of executions, the total amount of time of executions, the total number of executions currently in progress, and the number of failed executions.
+pipelines._id_.response.count | Integer | The number of search response processor executions by the search pipeline.
```
Comment (natebower) [inline _api-reference/nodes-apis/nodes-stats.md:1027]:
```suggestion
pipelines._id_.response.count | Integer | The number of search response processor executions performed by the search pipeline.
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Just three comments :-)

**Pull request merged**